app/src/androidTest/java/com/android/shelflife/ui/overview/AddFoodItemScreenTest.kt

```diff
@@ -336,4 +410,93 @@ class AddFoodItemScreenTest {
+          foodItemViewModel = foodItemViewModel,
+          foodFactsViewModel = foodFactsViewModel)
+    }
+
+    // Scroll to the image selection section
+    composeTestRule.onNodeWithTag("inputFoodName").performTextInput("Bananas")
+    composeTestRule
+        .onNodeWithTag("addFoodItemScreen")
+        .performScrollToNode(hasTestTag("selectImage"))
+    composeTestRule.onNodeWithTag("selectImage").assertIsDisplayed()
+    // Select the first image
+    composeTestRule.onAllNodesWithTag("foodImage")[0].performClick()
+
+    // Verify that the selected image is displayed
+  }
+
+  @Test
+  fun testNoImageOptionSelection() {
+    composeTestRule.setContent {
+      AddFoodItemScreen(
+          navigationActions = navigationActions,
+          houseHoldViewModel = houseHoldViewModel,
+          foodItemViewModel = foodItemViewModel,
+          foodFactsViewModel = foodFactsViewModel)
+    }
+
+    // Scroll to the "No Image" option
+    composeTestRule.onNodeWithTag("addFoodItemScreen").performScrollToNode(hasTestTag("noImage"))
+
+    // Select the "No Image" option
+    composeTestRule.onNodeWithTag("noImage").performClick()
+    // Verify that the default image is displayed
+  }
+
+  fun testDateReValidation() {
+    composeTestRule.setContent {
+      AddFoodItemScreen(
+          navigationActions = navigationActions,
+          houseHoldViewModel = houseHoldViewModel,
+          foodItemViewModel = foodItemViewModel,
+          foodFactsViewModel = foodFactsViewModel)
+    }
+
+    // Enter valid buy date
+    composeTestRule.onNodeWithTag("inputFoodBuyDate").performTextInput("01012026")
+
+    // Enter valid expire date
+    composeTestRule.onNodeWithTag("inputFoodExpireDate").performTextInput("31122026")
+
+    // Enter valid open date
+    composeTestRule.onNodeWithTag("inputFoodOpenDate").performTextInput("01022026")
+
+    // Change buy date to an invalid one
+    composeTestRule.onNodeWithTag("inputFoodBuyDate").performTextClearance()
+    composeTestRule.onNodeWithTag("inputFoodBuyDate").performTextInput("31132023")
+
+    // Verify error messages are displayed for dependent dates
+    composeTestRule.onNodeWithText("Invalid date").assertIsDisplayed()
+  }
+
+  inner class FakeFoodFactsRepository : FoodFactsRepository {
+    var shouldReturnError = false
+    var foodFactsList = listOf<FoodFacts>()
+
+    override fun searchFoodFacts(
+        searchInput: FoodSearchInput,
+        onSuccess: (List<FoodFacts>) -> Unit,
+        onFailure: (Exception) -> Unit
+    ) {
+      if (shouldReturnError) {
+        onFailure(Exception("Test exception"))
+      } else {
+        onSuccess(foodFactsList)
+      }
+    }
+  }
+
+  companion object {
+    const val DEFAULT_IMAGE_URL =
+        "https://media.istockphoto.com/id/1354776457/vector/default-image-icon-vector-missing-picture-page-for-website-design-or-mobile-app-no-photo.jpg?s=612x612&w=0&k=20&c=w3OW0wX3LyiFRuDHo9A32Q0IUMtD4yjXEvQlqyYk9O4="
+  }
```

Comment:
```suggestion
```
You can access the DEFAULT_IMAGE_URL for FoodFacts from its companion object without this new object.
Simply import it from here:
 `import com.android.shelfLife.model.foodFacts.FoodFacts.Companion.DEFAULT_IMAGE_URL`